sound/soc/sof/mediatek/mt8195/mt8195.c

```diff
@@ -278,6 +289,19 @@ static int mt8195_dsp_probe(struct snd_sof_dev *sdev)
 	sdev->mmio_bar = SOF_FW_BLK_TYPE_SRAM;
 	sdev->mailbox_bar = SOF_FW_BLK_TYPE_SRAM;
 
+	priv->adsp2ap_addr = dram_remap_to_ap;
+	priv->ap2adsp_addr = dram_remap_to_dsp;
+	share_dram.phy_addr = priv->adsp->pa_shared_dram;
+	share_dram.va_addr = (unsigned long long)priv->adsp->shared_dram;
+	share_dram.vir_addr = (unsigned char *)priv->adsp->shared_dram;
+	share_dram.size = SIZE_SHARED_DRAM_UL + SIZE_SHARED_DRAM_DL;
+	ret = adsp_genpool_create(&priv->mem_pool, &share_dram);
+	if (ret) {
+		dev_err(sdev->dev, "adsp_genpool_create fail!\n");
+		ret = -ENOMEM;
+		goto err_adsp_sram_power_off;
```

Comment:
this goto seems unnecessary. Is the intention to power off the adsp sram in case of a probe success also?